cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -753,14 +794,36 @@ def handle_etcd_relation(reldata):
     api_opts.add('etcd-servers', connection_string, strict=True)
 
 
-def configure_master_services():
-    ''' Add remaining flags for the master services and configure snaps to use
-    them '''
-
+def get_config_args():
+    db = unitdata.kv()
+    old_config_args = db.get('api-extra-args', [])
+    # We have to convert them to tuples becuase we use sets
+    old_config_args = [tuple(i) for i in old_config_args]
+    new_config_args = []
+    new_config_arg_names = []
+    for arg in hookenv.config().get('api-extra-args', '').split():
+        new_config_arg_names.append(arg.split('=', 1)[0])
+        if len(arg.split('=', 1)) == 1:  # handle flags ie. --profiling
```

Comment:
They are supported by kube-apiserver, but not `snap set kube-apiserver`. It takes them as boolean values instead, e.g. `contention-profiling=true` and `contention-profiling=false`.

My specific concern here is that this case will break on the following line:
```
new_config_args.append(tuple(arg.split('=', 1), None))
```

I don't think `tuple` accepts 2 arguments. That and the `None` probably needs to be `'true'` since it's eventually gonna go through FlagManager to a `snap set` call.

More generally, I get the feeling this case shouldn't exist at all. The description in config.yaml makes me think api-extra-args follows the same conventions as `snap set`, with the `profiling=true` example, but the case here says otherwise.